substrate/client/executor/polkavm/src/lib.rs

```diff
@@ -205,26 +204,32 @@ fn call_host_function(
 		Ok(value) => value,
 		Err(error) => {
 			log::warn!("Call into the host function '{}' failed: {error}", function.name());
-			return Err(polkavm::Trap::default());
+			return Err(polkavm::CallError::Trap);
```

Comment:
Now that `polkavm::CallError` has a `User` variant this should return a proper error message instead of `CallError::Trap`. (This previously returned `polkavm::Trap` only because there was no other choice.)